### PR TITLE
refactor: move fixtures to corresponding directories

### LINKED ISSUE #3328: Move test fixtures closer to tests
Currently all test fixtures are in tests/_fixtures/pypi while most are used within directories like tests/datasource/pypi. Pypi's fixtures should be moved to tests/datasource/pypi/_fixtures, for example. Need to make sure they are skipped by linters, etc.

### PR DESCRIPTION
**Pull request by rishabh1403**
- Refactor: move fixtures to corresponding directories
- Delete unused fixtures

Resolves: #3328

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1 +1,7 @@
 test/_fixtures
+test/datasource/**/_fixtures
+test/manager/**/_fixtures
+test/platform/**/_fixtures
+test/config/**/_fixtures
+test/workers/**/_fixtures
+test/logger/**/_fixtures
```
Comment (rarkins) [inline .eslintignore]: This is fine if we need, but is there any simpler pattern that just lets us ignore any `_fixtures` directory no matter where it is?

File: .gitignore [hunk 2]
```diff
@@ -8,6 +8,5 @@
 /*.log
 /.vscode
 /.idea
-package-lock.json
```
Comment (rarkins) [inline .gitignore]: This is to prevent someone accidentally adding `package-lock.json` to a PR. Is there a reason to remove it?

Comment (rishabh1403) [reply .gitignore]: One of the Fixtures had a package-lock.json, tarvis tests were failing because this wasn't getting committed to Github

Comment (rarkins) [reply .gitignore]: OK so now you have committed it, you should be able to revert this change and everything still passes. Alternatively you can change it to `/package-lock.json` and it should only affect the one in root.

Comment (rishabh1403) [reply .gitignore]: I guess changing it to `/package.json` would be better, this is to make sure no one accidentally deletes the required `package-lock.json` in the  `_fixtures`

File: .eslintignore (hunk 1 above)
Comment (rishabh1403) [reply .eslintignore]: @rarkins Replaced with a glob that will match any `_fixtures` directory in test directory

**Pull request merged**